### ISSUE TITLE
`{}` in paths will cause a failure to save or load

### ISSUE THREAD

**Issue by 5HT2** [anegostudios/VintageStory-Issues#1825]
**Game Version:** 1.16.4
**Platform:** **Windows** / Mac / Linux (which?)   
**Modded:** **No** / Yes (which?)  
**SP/MP:** **Singleplayer** / **Multiplayer** (which?)  

This is most noticable when first creating a world, as shown by the logs:

```
23.3.2022 16:53:45 [Notification] Server logger started.
23.3.2022 16:53:45 [Notification] Game Version: v1.16.4 (Stable)
23.3.2022 16:53:45 [Event] Launching server...
23.3.2022 16:53:45 [Event] Server v1.16.4, network v1.16.5, api v1.7.0
23.3.2022 16:53:45 [Notification] Entering runphase Initialization
23.3.2022 16:53:45 [Event] Loading configuration...
23.3.2022 16:53:45 [Notification] Entering runphase Configuration
23.3.2022 16:53:45 [Notification] serverconfig.json not found, creating new one
23.3.2022 16:53:46 [Notification] Loading savegame
23.3.2022 16:53:46 [Notification] Create new save game data. Playstyle: surviveandbuild
23.3.2022 16:53:46 [Notification] Default spawn was set in serverconfig, resetting for safety.
23.3.2022 16:53:46 [Error] Couldn't write to log file, failed formatting Savegame C:\Users\ReDDeZ_{Mark}\AppData\Roaming\VintagestoryData\Saves\vintage village lands.vcdbs loaded (FormatException)
23.3.2022 16:53:46 [Notification] Server stop requested, begin shutdown sequence. Stop reason: Exception thrown by server during startup or process
23.3.2022 16:53:46 [Notification] Entering runphase Shutdown
23.3.2022 16:53:46 [Notification] Defragmented listener lists
23.3.2022 16:53:46 [Event] Shutting down 5 server threads... 
23.3.2022 16:53:46 [Event] Killed console thread
23.3.2022 16:53:47 [Event] All threads gracefully shut down
23.3.2022 16:53:47 [Event] Doing last tick...
23.3.2022 16:53:47 [Event] Stopped the server!
23.3.2022 19:49:50 [Event] Launching server...
23.3.2022 19:49:50 [Event] Server v1.16.4, network v1.16.5, api v1.7.0
23.3.2022 19:49:50 [Notification] Entering runphase Initialization
23.3.2022 19:49:50 [Event] Loading configuration...
23.3.2022 19:49:50 [Notification] Entering runphase Configuration
23.3.2022 19:49:50 [Notification] Loading savegame
23.3.2022 19:49:50 [Notification] Loaded existing save game data. Playstyle: surviveandbuild, Playstyle Lang code: preset-surviveandbuild, WorldType: preset-surviveandbuild
23.3.2022 19:49:50 [Notification] Server stop requested, begin shutdown sequence. Stop reason: Exception thrown by server during startup or process
23.3.2022 19:49:50 [Notification] Entering runphase Shutdown
23.3.2022 19:49:50 [Notification] Defragmented listener lists
23.3.2022 19:49:50 [Error] Exception thrown during handling event OnGameWorldBeingSaved: System.NullReferenceException: Ссылка на объект не указывает на экземпляр объекта.
   в Vintagestory.Server.ServerSystemWorldAmbient.OnSaving()
   в Vintagestory.Server.ServerEventManager.TriggerGameWorldBeingSaved()
23.3.2022 19:49:50 [Event] Shutting down 5 server threads... 
23.3.2022 19:49:50 [Event] Killed console thread
23.3.2022 19:49:50 [Event] All threads gracefully shut down
23.3.2022 19:49:50 [Event] Doing last tick...
23.3.2022 19:49:50 [Event] Stopped the server!
```

The lines to look out for are 

> `Couldn't write to log file, failed formatting Savegame C:\Users\ReDDeZ_{Mark}\AppData\Roaming\VintagestoryData\Saves\vintage village lands.vcdbs loaded (FormatException)`

and

> `Ссылка на объект не указывает на экземпляр объекта.`

The exception to get to the root of our problem is here ([log](https://cdn.discordapp.com/attachments/334233095374897152/956203930646351912/client-main.txt)):

```
23.3.2022 16:53:46 [Fatal] System.FormatException: Входная строка имела неверный формат.
   в System.Text.StringBuilder.FormatError()
   в System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   в System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   в System.String.Format(IFormatProvider provider, String format, Object[] args)
   в System.IO.TextWriter.WriteLine(String format, Object[] arg)
   в System.IO.TextWriter.SyncTextWriter.WriteLine(String format, Object[] arg)
   в System.Console.WriteLine(String format, Object[] arg)
   в _VYGBNydfJ6NqUARtkpE4d5DQXJD._ngYW4wW0dY4aQCzsws72ReRBAJO(EnumLogType , String , Object[] )
   в Vintagestory.Server.ServerLogger.LogImpl(EnumLogType logType, String message, Object[] args)
   в Vintagestory.API.Common.LoggerBase.Log(EnumLogType logType, String format, Object[] args)
   в Vintagestory.Server.ServerSystemLoadAndSaveGame.LoadSaveGame()
   в Vintagestory.Server.ServerSystemLoadAndSaveGame.OnBeginConfiguration()
   в Vintagestory.Server.ServerMain.EnterRunPhase(EnumServerRunPhase runPhase)
   в Vintagestory.Server.ServerMain.Launch()
   в _TRhArnJABwZoV9Vt7Ld2l65Xzgg._55FJfeMcDB5NT6lsb5fDE3bzUPI()
```
   
> `Входная строка имела неверный формат.`
   
This translates to "Input string was not in a correct format". This implies that there's something wrong with the path, i.e. having `{` or `}` in it.
   
The stackoverflow post describing this issue can be found at: https://stackoverflow.com/q/14856648.

The noted solution is replacing the `{` with a `{{`, and so on, that will allow the string to be parsed properly.

A temporary workaround that has been confirmed to work is removing the `{` from the path (i.e. changing your username).

**Comment by copygirl** [comment]
This was an issue I found with the game's logger implementation, but my suggested fix was overlooked because I wasn't able to compile the game at the time on my system

**Comment by DArkHekRoMaNT** [comment]
Issue should be fixed in 1.18.0
> Fixed: Game crashing on world load when the installation path contained curly braces
